patroni/postgresql/slots.py

```diff
@@ -417,8 +417,20 @@ def _ensure_physical_slots(self, slots: Dict[str, Any], clean_inactive_physical_
             # And advance restart_lsn on physical replication slots that are not expected to be active.
             elif self._postgresql.can_advance_slots and self._replication_slots[name]['type'] == 'physical' and\
                     value.get('expected_active') is not True and not value['xmin']:
+                restart_lsn = value.get('restart_lsn')
+                if not restart_lsn:
+                    # This slot either belongs to a member or was configured as a permanent slot. However, for some
+                    # reason the slot was created by an external agent instead of by Patroni, and it was created without
+                    # reserving the LSN. We ignore the slot here, as we cannot advance it. The user might want to drop
+                    # the slot and let Patroni recreate and manage it.
+                    logger.warning(
+                        "Physical replication slot '%s' has no restart_lsn, cannot advance it. "
```

Comment:
Shouldn't we give users more context, what they should do?
Maybe take a step forward and delete this slot right away?

Reply:
Maybe we can add information that they might want to drop the slots?

I was tempted to go ahead and remove, but I wonder if we should always take that action, or leave the option to the user to check if they did some mistake?

I guess the latter would be a bit more conservative 😆

Reply:
Lgtm. But what would be your arguments against deletion?

Reply:
I've been thinking about it, and I guess we can drop the slot on behalf of the user.

I was only afraid of wrongly taking some destructive action. But in theory, if this code is reached, that's either a permanent or a member physical slot, which are expected to be managed by Patroni. So I think we are safe to drop it through Patroni.

I can push a commit with that change later if you are also inclined to that approach :)

Reply:
I would vote for it:) @CyberDem0n 

Btw, how did that happen in your case? Users thought they need to do it for Patroni or this slot was used for some other use case?